src/Configurator.php

```diff
@@ -2,34 +2,135 @@
      */
-    public function setContainer($container);
+    public static function apply()
+    {
+        return new self();
+    }
+
+    private function __construct()
+    {
+        $this->config = new ApplicationConfig([]);
+    }
 
     /**
-     * @param ApplicationConfig $config
-     * @param string            $prefix
+     * @api
      *
-     * @return void
+     * @param array $config
+     *
+     * @return Configurator
      */
-    public function addApplicationConfig(ApplicationConfig $config, $prefix = 'config');
+    public function configFromArray(array $config)
+    {
+        $this->config->merge($config);
+
+        return $this;
+    }
 
     /**
-     * @param ServiceConfig $config
+     * @api
      *
-     * @return void
+     * @param string $pattern
+     *
+     * @return Configurator
+     */
+    public function configFromFiles($pattern)
+    {
+        $locator = new FileLocator();
+
+        $factory = new ReaderFactory([
+            '.json' => 'TomPHP\ConfigServiceProvider\FileReader\JSONFileReader',
+            '.php'  => 'TomPHP\ConfigServiceProvider\FileReader\PHPFileReader',
+        ]);
+
+        $files = $locator->locate($pattern);
+
+        if (empty($files)) {
+            throw NoMatchingFilesException::fromPattern($pattern);
+        }
+
+        foreach ($files as $filename) {
+            $reader = $factory->create($filename);
+            $this->config->merge($reader->read($filename));
+        }
+
+        return $this;
+    }
+
+    /**
+     * @api
+     *
+     * @param string $name
+     * @param mixed  $value
+     *
+     * @return Configurator
      */
-    public function addServiceConfig(ServiceConfig $config);
+    public function withSetting($name, $value)
+    {
+        if (!array_key_exists($name, $this->settings)) {
+            throw UnknownSettingException::fromSetting($name, array_keys($this->settings));
+        }
+
+        $this->settings[$name] = $value;
+
+        return $this;
+    }
 
     /**
-     * @param InflectorConfig $config
+     * @api
+     *
+     * @param object $container
      *
      * @return void
      */
-    public function addInflectorConfig(InflectorConfig $config);
+    public function to($container)
+    {
+        $this->config->setSeparator($this->settings['config_separator']);
+
+        $factory = new ConfiguratorFactory([
+            'League\Container\Container' => 'TomPHP\ConfigServiceProvider\League\LeagueContainerAdapter',
+            'Pimple\Container'           => 'TomPHP\ConfigServiceProvider\Pimple\PimpleContainerAdapter',
+        ]);
+
+        $configurator = $factory->create($container);
+
+        $configurator->addApplicationConfig($this->config, $this->settings['config_prefix']);
+
+        if (isset($this->config[$this->settings['services_key']])) {
+            $configurator->addServiceConfig(
+                new ServiceConfig($this->config[$this->settings['services_key']], $this->settings['singleton_services'])
```

Comment:
@tomphp 

Since `$config` is technical a single argument to `addServiceConfig()`, _but_ the constructor of `ServiceConfig` accepts two arguments, should we wrap its arguments instead?

```php
$configurator->addServiceConfig(new ServiceConfig(
    $this->config[$this->settings['services_key']],
    $this->settings['singleton_services']
));
```

What do you think?